table/data_block_hash_index.cc

```diff
@@ -8,49 +8,37 @@
 #include "rocksdb/slice.h"
 #include "table/data_block_hash_index.h"
 #include "util/coding.h"
-#include "util/hash.h"
+#include "util/xxhash.h"
 
 namespace rocksdb {
 
 const uint32_t kSeed = 2018;
-const uint32_t kSeed_tag = 214; /* second hash seed */
 
 inline uint16_t HashToBucket(const Slice& s, uint16_t num_buckets) {
   return static_cast<uint16_t>(
-      rocksdb::Hash(s.data(), s.size(), kSeed) % num_buckets);
+      rocksdb::XXH32(s.data(), static_cast<int>(s.size()), kSeed) % num_buckets);
```

Comment:
Still try to use Hash. What is kSeed and why do we need it?

Reply:
I figured. I think it's better to directly use GetSliceHash().

Reply:
GetSliceHash() still uses util/hash.cc, i.e. rocksdb::Hash(), the original problem still exists and I verified by small experiment. In the experiment, where the Hash value mod 4096, resulting in no more than 32 possible results:
```
#key     # hash_values  #hash_values
...
num_key=249 XXH32=244 GetSliceHash=32
num_key=249 XXH32=237 GetSliceHash=32
num_key=249 XXH32=243 GetSliceHash=32
num_key=249 XXH32=241 GetSliceHash=32
num_key=249 XXH32=244 GetSliceHash=32
...
```
Besides, when not using a buckets of power of 2, some cases the hash collision is unexpectedly high:
```
GetSliceHash
util_ratio=1.0 collision_ratio=0.5459249
util_ratio=0.9 collision_ratio=0.8764291 <===
util_ratio=0.8 collision_ratio=0.6734214
util_ratio=0.7 collision_ratio=0.606063
util_ratio=0.6 collision_ratio=0.4476331
util_ratio=0.5 collision_ratio=0.3601262
util_ratio=0.4 collision_ratio=0.3359398
util_ratio=0.3 collision_ratio=0.6172485 <===
util_ratio=0.2 collision_ratio=0.1536234
util_ratio=0.1 collision_ratio=0.1480481

XXHash
util_ratio=1.0 collision_ratio=0.6315011
util_ratio=0.9 collision_ratio=0.5931281
util_ratio=0.8 collision_ratio=0.5496543
util_ratio=0.7 collision_ratio=0.5028827
util_ratio=0.6 collision_ratio=0.4510745
util_ratio=0.5 collision_ratio=0.3923727
util_ratio=0.4 collision_ratio=0.328849
util_ratio=0.3 collision_ratio=0.2584299
util_ratio=0.2 collision_ratio=0.1810817
util_ratio=0.1 collision_ratio=0.094881
```
May be we can still use `GetSliceHash()` and only treat the case of power of 2 buckets specially, thought there may still be some random occurences of high hash collision.

Reply:
I am not exactly sure about seeds. Seen from the impelmentation, it determines an initial state of the temp value. 

I think with different seed, it behaves like different hash function.